lib/goodcity/detail_factory.rb

```diff
@@ -0,0 +1,94 @@
+module Goodcity
+  class DetailFactory
+    PERMITTED_DETAIL_TYPES = %w[computer electrical computer_accessory].freeze
+    FIXED_DETAIL_ATTRIBUTES = %w[comp_test_status test_status frequency voltage].freeze
+
+    attr_accessor :item, :package, :stockit_detail_id
+
+    def initialize(item, package)
+      @item = item
+      @package = package
+      @stockit_detail_id = item["detail_id"]
+    end
+
+    def run
+      import_and_save_detail? && !detail_present? && package_updated?
+    end
+
+    private
+
+    def package_updated?
+      package.update_columns(detail_id: detail_id, detail_type: detail_type)
+    end
+
+    def import_and_save_detail?
+      item["id"] && item["detail_type"] && stockit_detail_id
+    end
+
+    def detail_present?
+      package.detail.present?
```

Comment:
This is used only at one place [above](https://github.com/crossroads/api.goodcity/pull/849/files#diff-4595f7b4ee82ab803bd631af46579420R15) and also checking the absence of `package.details`. Better to remove it and directly use as `package.detail.blank?`